### DISCUSSION TITLE
Make 2 elements scroll together

### DISCUSSION THREAD

**Discussion by ThatOneCalculator** [idiotWu/smooth-scrollbar#460] [Q&A] [answered]
I have 2 html elements that I want to scroll *together* whenever the scroll wheel is touched, is there any way to link them up?

**Comment by dolphin-wood** [comment, marked as answer]
Assuming the two elements have exactly the same layout, you can try:

```js
one.addListener(({ offset }) => {
  two.setPosition(offset.x, offset.y);
});

// add this if necessary
two.addListener(({ offset }) => {
  one.setPosition(offset.x, offset.y);
});
```

**Reply by ThatOneCalculator** [reply]
They don't have the same layout, and when I tried that, it didn't scroll the 2nd element.

**Reply by dolphin-wood** [reply]
Can you provide a demo?

**Reply by ThatOneCalculator** [reply]
> Can you provide a demo?

As of right now, I've disabled smooth scrollbar on the page, but if you look at it, the line numbers and the text content are two separate elements and should scroll together.
https://stepbro.voring.me/~/f9SNJLQ

**Reply by dolphin-wood** [reply]
Thanks for the demo! Just wondering if it's possible to wrap the two components together into the same container, i.e.

```html
<Scrollbar>
  <LineNum />
  <Code />
</Scrollbar>
```

**Reply by ThatOneCalculator** [reply]
> Thanks for the demo! Just wondering if it's possible to wrap the two components together into the same container, i.e.
> 
> ```
> <Scrollbar>
>   <LineNum />
>   <Code />
> </Scrollbar>
> ```

Nah, when you do, the line numbers appear before the text vertically. 
I'm sure that could be fixed in the CSS, but I'm not the best front-end dev ^^"

**Reply by dolphin-wood** [reply]
**Fix 1: init scrollbar on `.wrapper` and apply flex layout to `.scroll-content`**

```css
.wrapper .scroll-content {
  display: flex;
}
```

**Fix 2: wrap the `.wrapper` into another element**

```html
<main class="scrollbar-container">
  <div class="wrapper"> ... </div>
</main>
```

```css
.scrollbar-container {
  height: 100%;
}

.wrapper {
  /* height: calc(100% - 2rem); */ <- remove this
}
```

```js
Scrollbar.init(document.querySelector('.scrollbar-container'))
```

**Reply by ThatOneCalculator** [reply]
Works great, thank you!
